Source/WebCore/Modules/webxr/WebXRWebGLLayer.h

```diff
@@ -110,7 +110,6 @@ class WebXRWebGLLayer : public WebXRLayer, private CanvasObserver {
     bool m_antialias { false };
     bool m_ignoreDepthValues { false };
     bool m_isCompositionEnabled { true };
-    bool m_viewportsDirty { true };
```

Comment:
@djg do we need this boolean? It wasn't getting updated when the viewports changed from the compositor and all it prevents is assignment of two MTLViewport instances, so its not preventing much work in the first place.

Reply:
@mwyrzykowski This is to do with the XR spec (which I'm no expert on), I think it's to do with the dynamic scaling feature in the spec.

Reply:
So the problem is that the underlying frameworks can change the viewport size with no signal from the JavaScript layer. How do we tell WebXRWebGLLayer that it needs to update the viewport in that case? Is it wrong to unconditionally update the viewports?